src/aleph/sdk/__init__.py

```diff
@@ -12,3 +12,14 @@
     del get_distribution, DistributionNotFound
 
 __all__ = ["AlephHttpClient", "AuthenticatedAlephHttpClient"]
+
+
+def __getattr__(name):
+    if name == "AlephClient":
+        raise ImportError(f"AlephClient has been turned into an abstract class. Please use `AlephHttpClient` instead.")
+    elif name == "AuthenticatedAlephClient":
+        raise ImportError(f"AuthenticatedAlephClient has been turned into an abstract class. Please use `AuthenticatedAlephHttpClient` instead.")
+    elif name == "synchronous":
+        raise ImportError(f"The 'aleph.sdk.synchronous' type is deprecated and has been removed from the aleph SDK. Please use `aleph.sdk.client.AlephHttpClient` instead.")
+    elif name == "asynchronous":
+        raise ImportError(f"The 'aleph.sdk.asynchronous' type is deprecated and has been removed from the aleph SDK. Please use `aleph.sdk.client.AlephHttpClient` instead.")
```

Comment:
Would it not be better to do a dictionary lookup ?

`ImportError` does not seem like the right exception here :thinking: 
[See https://docs.python.org/3/library/exceptions.html](https://docs.python.org/3/library/exceptions.html#exception-hierarchy)
`AttributeError`  is the standard class for this in Python, see 
https://docs.python.org/3/reference/datamodel.html#object.__getattr__

Why use f-strings without any formatting ?

```suggestion
        if name in DEPRECATION_MESSAGES:
            raise AttributeError(DEPRECATION_MESSAGES[name])
        else:
            return super().__getattr__(name)
```